bin/resty

```diff
@@ -360,13 +360,23 @@ if ($use_valgrind) {
 
 my $child_pid;
 
-sub sigint {
-    $SIG{INT} = \&sigint;
+sub forward_signal {
+    my $signame = shift;
+
     if ($child_pid) {
-        kill INT => $child_pid;
+        unless(kill $signame => $child_pid) {
+            die "child process is dead, can't forward signal to it";
```

Comment:
Didn't saw the `$!` in [kill](http://perldoc.perl.org/functions/kill.html). Maybe in somewhere
else, need to keep digging :). I am quite new to perl.